### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by hexcoder-**

### CODE REVIEW

**Comment by domenukk** [comment]
I don't get this PR, we did a merge to stable 2 days ago, yet th PR contains some commits that are 7+ days old? How?

**Comment by hexcoder-** [comment]
@domenukk: this is my work from the going_atomic branch, which first got merged into dev, then merged to stable.

File: instrumentation/afl-llvm-pass.so.cc [hunk 1]
```diff
@@ -715,6 +774,117 @@ bool AFLCoverage::runOnModule(Module &M) {
 
     }
 
+    if (use_threadsafe_counters) {                       /*Atomic NeverZero */
+      // handle the list of registered blocks to instrument
+      for (auto val : todo) {
```
Comment (domenukk) [inline instrumentation/afl-llvm-pass.so.cc:785]: Is this todo variable .. fine here?

File: docs/Changelog.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ sending a mail to <afl-users+subscribe@googlegroups.com>.
       it fails
   - afl-cc:
     - We do not support llvm versions prior 6.0 anymore
+    - added thread safe counters to all modes (`AFL_LLVM_THREADSAFE_INST`),
+      note that this disables never zero counters.
```
Comment (domenukk) [inline docs/Changelog.md]: NeverZero ?

Comment (hexcoder-) [reply docs/Changelog.md]: Yes, will change that.

File: instrumentation/afl-llvm-pass.so.cc (hunk 1 above)
Comment (hexcoder-) [reply instrumentation/afl-llvm-pass.so.cc:785]: Yes, it is just a name. It contains the parts, where new blocks are to be inserted.

**Review by domenukk — APPROVED** [review]

**Pull request merged**